Allows the bridge to be in a separate server not just localhost

gastonmorixe opened the pull request:
> [!CAUTION]
> This is not a recommended change nor use

For very few cases and advanced users or devs, this changes the hard-coded `127.0.0.1` IP from multiple parts exposing the bridge to the whole network allowing connections from anywhere by changing the hard-coded IP to `0.0.0.0`. 

**Do not use**. **Do not merge.** It's just a reference for anyone who may need it for advanced or special cases. 

Then build it with:
```bash
$ make build-nogui
```

ilmc888 commented:
Might be nice to make this configurable? Can be hidden away as advanced option to avoid accidents, but would be more convenient than compiling and keeping the patch up to date.

I have been playing with systemd socket to forward requests, so my paperless container can connect to it, but stranded at a mismatch of the hostname in the certificate (127.0.0.1 vs host.containers.internal).

waTeim commented:
Yea, just add  `--bind <ip>`;  it's for containers, kubernetes, agents.  Very soon the majority will not care about the UI.

gastonmorixe committed:
feat: bind to all network interfaces (0.0.0.0) instead of localhost

Allow Proton Bridge to be accessed from remote machines on the network
by changing all 127.0.0.1 bindings to 0.0.0.0. This affects IMAP/SMTP
listeners, gRPC services, TLS certificate generation, the Focus service,
and the GUI frontend connections.

Also update .gitignore to exclude local TLS certificates, build
artifacts, service scripts, and research docs.

gastonmorixe committed:
feat: bind to all network interfaces (0.0.0.0) instead of localhost

Allow Proton Bridge to be accessed from remote machines on the network
by changing all 127.0.0.1 bindings to 0.0.0.0. This affects IMAP/SMTP
listeners, gRPC services, TLS certificate generation, the Focus service,
and the GUI frontend connections.

Also update .gitignore to exclude local TLS certificates, build
artifacts, service scripts, and research docs.

pkolbus commented:
FWIW, I've solved this by setting up a couple of socat units as described here: https://vimoire.com/blog/2025/setting_up_protonmail_bridge_on_lan_server -- this also allows using the standard port numbers (e.g. 993 for IMAPS). No patching of proton-bridge needed.

An SSL cert of your choice can be installed with `protonmail-bridge --cli` commands -- I'm using a LetsEncrypt wildcard cert to avoid iOS trust issues, but a self-signed cert (for whatever name the server goes by) should also work if your client allows.
